### PR TITLE
feat(storage-account): remove data block

### PR DESCRIPTION
**Pull request by filipefigueredo**
# Description

The idea of this PR was to remove data blocks, since we found issues using them if any data changes.

# Motivation and Context

As we mentioned last week, when using data blocks, we have the issue of needing to employ 'depends on' statements, which means that every time data changes, the resource will be removed and recreated rather than just updated.

This PR corrects that by delegating the module's location and tags to whoever will use it. (in other words: added location and tags as variables.) 

Also:
- Updated variable descriptions to conform with the ones already used in other modules. (trying to follow the resource-group descriptions)
- Fixed `azurerm` provider version to 2.98.0 (ensuring that it will operate consistently, rather than relying on Azure developers to do the right thing with versioning)
- Enforced TLS 1.2
- Enforced HTTPS-only traffic
- Created validations for the inputs following the `azurerm_ storage_account` [documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account):

# How Has This Been Tested?

1. Create a new folder in your computer and copy the storage_account folder to it
2. Create a new tf file (main.tf for example) and add the code below referencing the 'storage_account' module

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "=2.98.0"
    }
  }

  required_version = ">= 1.0.0"
}

provider "azurerm" {
  features {
  }
}
#########
variable "project" {
  description = "nmbrs project name."
  type        = string
  default     = "test-product"
}

variable "tags" {
  type        = map(string)
  description = "list of nmbrs mandatory resource tags."
  default = {
    country : "nl"
    squad : "infra"
    product : "test-product"
    environment : "dev"
  }

}

variable "environment" {
  type        = string
  description = "environment type of the apps"
  default     = "dev"
}

#########
module "rg" {
  source      = "git::github.com/Nmbrs/tf-modules//azure/resource-group?ref=main"
  environment = var.environment
  project     = var.project
  location    = "westeurope"
  tags        = var.tags
}

module "storage_account" {
  source = "./azure/storage_account"

  name                = "ffssanmbrstest001"
  location            = "West Europe"
  resource_group_name = module.rg.resource_group_name
  tags = {
    my_tag : "my_tag_value"
  }
  account_kind     = "Storage"
  account_tier     = "Standard"
  replication_type = "GRS"
}
```
3. terraform init
4. terraform validate
5. terraform plan

# Screenshots

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [ ] My code has security implications.
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: azure/storage_account/README.md [hunk 1]
```diff
@@ -16,9 +16,13 @@ Fundamentally, you need to declare the module and pass the following variables i
 ```hcl
```
Comment (fargozhu) [inline azure/storage_account/README.md:16]: didn't know that markdown was supporting HCL language 👍🏻

File: azure/storage_account/terraform.tf [hunk 2]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "~> 2.97.0"
+      version = "=2.98.0"
```
Comment (fargozhu) [inline azure/storage_account/terraform.tf:5]: shouldn't we move the provider version to the service scope?

File: azure/storage_account/variables.tf [hunk 3]
```diff
@@ -1,49 +1,61 @@
...
 variable "project" {
   type        = string
-  description = "nmbrs project name."
+  description = "Project name for the resource."
 }
 
-variable "environment" {
+variable "location" {
+  # For a complete list of available Azure regions run at cli:  
+  # az account list-locations  --query "[].{displayName:displayName, location:name}" --output table
```
Comment (fargozhu) [inline azure/storage_account/variables.tf:13]: I think this is not necessary and only brigs noise to the readability.

File: azure/storage_account/variables.tf [hunk 4]
```diff
@@ -1,49 +1,61 @@
...
+
+variable "tags" {
+  description = "nmbrs list of mandatory resource tags."
+  type        = map(string)
 }
 
 variable "kind" {
   # Valid options are BlobStorage, BlockBlobStorage, FileStorage, Storage and StorageV2. 
-  type        = string
   description = "Defines the Kind of storage account."
```
Comment (fargozhu) [inline azure/storage_account/variables.tf:30]:
```suggestion
  description = "Defines the Kind of storage account (options are BlobStorage, BlockBlobStorage, FileStorage, Storage and StorageV2)"
```

File: azure/storage_account/variables.tf [hunk 5]
```diff
@@ -1,49 +1,61 @@
...
+  type        = string
   default     = "StorageV2"
 }
 
 variable "account_tier" {
   # Valid options are Standard and Premium. 
   # For BlockBlobStorage and FileStorage accounts only Premium is valid.
-  type        = string
   description = "Defines the Tier to use for this storage account."
```
Comment (fargozhu) [inline azure/storage_account/variables.tf:38]:
```suggestion
  description = "Defines the Tier to use for this storage account (options are Standard and Premium)."
```

File: azure/storage_account/variables.tf [hunk 6]
```diff
@@ -1,49 +1,61 @@
...
 variable "replication_type" {
   # Valid options are LRS, GRS, RAGRS, ZRS, GZRS and RAGZRS.
-  type        = string
   description = "Defines the type of replication to use for this storage account."
+  type        = string
   default     = "LRS"
 }
 
 variable "min_tls_version" {
   # Valid options are TLS1_0, TLS1_1, and TLS1_2.
```
Comment (fargozhu) [inline azure/storage_account/variables.tf:51]: these aren't valid options. we should enforce tls1_2

**Review by fargozhu — COMMENTED** [review]
I did a quick test with my heimdall dummy service using the pr test guidelines and it worked accordingly with the specs.

All the review comments are just nice to have, so it doesn't block the pr.

File: azure/storage_account/variables.tf (hunk 4 above)
Comment (filipefigueredo) [reply azure/storage_account/variables.tf:22]: Fixed by adding validation blocks

**Pull request merged**